bindings/rust/wasmedge-sys/src/instance/table.rs

```diff
@@ -25,11 +25,109 @@ impl Table {
             }),
         }
     }
+
+    /// Get the table type from a table instance.
+    pub fn get_type(&self) -> WasmEdgeResult<TableType> {
+        let ty_ctx = unsafe { wasmedge::WasmEdge_TableInstanceGetTableType(self.ctx) };
+        match ty_ctx.is_null() {
+            true => Err(Error::OperationError(String::from(
+                "fail to get type info from the Table instance",
+            ))),
+            false => Ok(TableType {
+                ctx: ty_ctx as *mut _,
+                registered: true,
+            }),
+        }
+    }
+
+    /// Get the reference value in a table instance.
+    pub fn get_data(&self, offset: usize) -> WasmEdgeResult<Value> {
+        let ptr_raw_val = std::ptr::null_mut();
+        unsafe {
+            check(wasmedge::WasmEdge_TableInstanceGetData(
+                self.ctx,
+                ptr_raw_val,
+                offset as u32,
+            ))?;
+        }
+        let ref_raw_val = unsafe { ptr_raw_val.as_ref().unwrap() };
+        let raw_val = ref_raw_val.to_owned();
+        Ok(raw_val.into())
+    }
+
+    /// Set the reference value into a table instance.
+    pub fn set_data(&mut self, data: Value, offset: usize) -> WasmEdgeResult<()> {
+        unsafe {
+            check(wasmedge::WasmEdge_TableInstanceSetData(
+                self.ctx,
+                data.into(),
+                offset as u32,
+            ))
+        }
+    }
+
+    /// Get the size of a table instance.
+    pub fn get_size(&self) -> u32 {
```

Comment:
It may be good to provide `fn len(&self) -> usize` as a shortcut for `get_size`.
Or implement all of this in other PR.

Reply:
Accepted. I'll refactor this interface in this PR, and deal with similar ones in other PR. In addition, do we need to replace each `u32` with `usize`? As you may know, `usize` is pointer-based and depends on hardware. From the view of the library design, `usize` could be more reasonable than `u32`. How do you think about the choice between `usize` and `u32` as the return value type for wasmedge-sys.

Reply:
The function `len` should return `usize`, such that users does not need to transfer this, and we do the transfer in the framework, and keep it simple and stupid for users.

Reply:
Accepted. Will follow the comment, thanks a lot!

Reply:
We can keep `get_size` function, and we need a `len` function for this.
Just like other `len` function in std.
https://doc.rust-lang.org/stable/std/?search=len

`pub fn len(&self) -> usize`

It is good to use the `len` for the function name, not `size`.
Such that rust developer can write correct code without reading doc.
That is the ideal.

Reply:
Sorry about this. 
It may be good to use `capacity` for the size of the table, because the memory is not fill with data.

`len` is for data, and the `capacity` is for the size of memory.

```pub fn capacity(&self) -> usize```
> Returns the number of elements the binary heap can hold without reallocating.

Because the data may not real set, the `capacity` is better.

Reply:
Accepted. `capacity` is a better choice. I'll refactor the signature, thanks!